codebuild/spec/buildspec_sanitizer.yml

```diff
@@ -63,6 +63,12 @@ batch:
         variables:
           S2N_LIBCRYPTO: openssl-1.0.2
           COMPILER: clang
+    - identifier: clang_openssl_1_0_2_fips
```

Comment:
What is the benefit of running these tests with both clang/gcc? 

I agree that it's important to have gcc and clang tests, but I don't think we want gcc/clang tests for each CI variable that we have.

Reply:
It actually really matters for AddressSanitizer. The capabilities vary between the compilers. When I was recently writing up example memory leaks for our internal training, some of them would be caught by one compiler but not the other :upside_down_face: